### PR TITLE
add legacy reward claim contracts

### PR DESCRIPTION
**Pull request by credence0x**

### CODE REVIEW

File: legacyClaim/starknet/src/claims.cairo [hunk 1]
```diff
@@ -0,0 +1,3006 @@
+fn claims_mapping(num: felt252) -> (felt252, felt252) {
+    match num {
+        0 => panic!("zero claims"), 
+        1 => (0xDB65702A9b26f8a643a31a4c84b9392589e03D7c, 0xD3C21BCECCEDA1000000),
```
Comment (milancermak) [inline legacyClaim/starknet/src/claims.cairo:4]: The values don't look right. It's the same owner addr and amount, repeated 3000 times.

File: legacyClaim/starknet/src/bridge.cairo [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+        assert!(req.owner_l1.into() == claim_l1_owner, "Bridge: l1 caller not reward owner");
+
+        // ensure claim can't be done more than once
+        assert!(self.claimed.read(req.owner_l1) == false, "Bridge: l1 caller already claimed");        
+        self.claimed.write(req.owner_l1, true);
+
+        // transfer reward
+        assert!(
+            ERC20ABIDispatcher{contract_address: self.reward_token.read()}
+            .transfer_from(self.reward_sponsor.read(), req.owner_l2, claim_amount.into()),
```
Comment (milancermak) [inline legacyClaim/starknet/src/bridge.cairo:111]: So the reward_sponsor will give unlimited allowance to the L2 bridge contract, to ensure this won't fail? Just asking, couldn't find this in a deploy script (not sure if it's part of the PR tbh).

File: legacyClaim/starknet/src/claims.cairo (hunk 1 above)
Comment (credence0x) [reply legacyClaim/starknet/src/claims.cairo:4]: they are placeholder

File: legacyClaim/starknet/src/bridge.cairo (hunk 2 above)
Comment (credence0x) [reply legacyClaim/starknet/src/bridge.cairo:111]: the approval can be limited to the total amount we expect to be claimed. havent added deploy script. will do so in a few

Comment (milancermak) [reply legacyClaim/starknet/src/bridge.cairo:111]: Agreed, total amount expected is safest.

**Pull request merged**